### PR TITLE
Add more state sensors for DomesticHotWaterProduction

### LINKED ISSUE #568: Add more state sensors for DomesticHotWaterProduction (io:AtlanticDomesticHotWaterProductionV2_CE_FLAT_C2_IOComponent)
### Did you read the instructions?

- [X] I have read the [README](https://github.com/iMicknl/ha-tahoma/blob/master/README.md), including the [advanced debugging](https://github.com/iMicknl/ha-tahoma/blob/master/README.md#advanced) section.

### The problem

The integration works well with cozytouch. I have a Thermor water heater which is well detected.
I can change mode etc

But an important data is missing. This data exists in the cozytouch custom component (https://github.com/Cyr-ius/hass-cozytouch). It is a boolean called energy_demand (io:OperatingModeCapabilitiesState energy_demand) which is set to true when water heater is heating.
I think it is beacause power is not retrieved from cozytouch data.
Some other data are missing like core:V40WaterVolumeEstimationState which is the hot water volume already consumed.

### What version of this integration (ha-tahoma) has the issue?

2.7.2

### What version of Home Assistant Core has the issue?

2021.9.6

### Which gateway / hub do you use?

CozyTouch

### Device model

DomesticHotWaterProduction

### Device type

E722804

### Additional information

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by iMicknl**
Fixes #568

<a href="https://gitpod.io/#https://github.com/iMicknl/ha-tahoma/pull/637"></a>

### CODE REVIEW

File: custom_components/tahoma/binary_sensor.py [hunk 1]
```diff
@@ -73,6 +73,13 @@
         device_class=binary_sensor.DEVICE_CLASS_VIBRATION,
         is_on=lambda state: state == STATE_DETECTED,
     ),
+    # DomesticHotWaterProduction/WaterHeatingSystem
+    OverkizBinarySensorDescription(
+        key="io:OperatingModeCapabilitiesState",
+        name="Energy Demand Status",
+        device_class=binary_sensor.DEVICE_CLASS_HEAT,
+        is_on=lambda state: state["energyDemandStatus"] == 0,
```
Comment (tetienne) [inline custom_components/tahoma/binary_sensor.py:81]: Hopefully, for the moment we retrieve only one value from the json. If we need to retrieve a second value, we will have to rethink the description key system.

**Review by tetienne — APPROVED** [review]

File: custom_components/tahoma/binary_sensor.py (hunk 1 above)
Comment (iMicknl) [reply custom_components/tahoma/binary_sensor.py:81]: Same...

**Comment by vlebourl** [comment]
Did you take a look at what #617 does? I'd like to merge it before we finish the enums as it fixes a bug in away mode and introduces boost mode (or the other way around).

**Comment by iMicknl** [comment]
> Did you take a look at what #617 does? I'd like to merge it before we finish the enums as it fixes a bug in away mode and introduces boost mode (or the other way around).

You can merge that if is works :). This one will just add a few optional sensors, I don't think it has duplicate functionality?

**Pull request merged**